### PR TITLE
Add firebase@5.4.0 w/ npm auto-aupdate

### PR DESCRIPTION
**Pull request by sashberd**
# Pull request template for new library

Moved from #8030 

Pull request for issue: #7795 
Related issue(s): # #10004 

Checklist for **Pull request** or **lib adding request issue** follows the conventions.

Note that if you are using a distribution purpose repository/package, please also provide the url and other related info like popularity of the source code repo/package.

## Profile of the lib

* Git repository (required): https://github.com/firebase/firebase-js-sdk
* Official website (optional, not the repository): https://firebase.google.com/docs/web/setup
* NPM package url (optional): https://www.npmjs.com/package/firebase
* License and its reference: Apache 2.0
* GitHub / Bitbucket popularity (required):
  * Count of watchers: 129
  * Count of stars: 1317
  * Count of forks: 182
* NPM download stats (optional):
  * Downloads in the last day:
  * Downloads in the last week: 467557
  * Downloads in the last month:

## Essential checklist

* [ ] I'm the author of this library
  * [ ] I would like to add link to the page of this library on CDNJS on website / readme
* [x] This lib was not found on cdnjs repo
* [x] Doesn't already exist / has duplicate issue and PR
* [x] The lib has notable popularity
  * [x] More than 200 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
  * [x] More than 800 downloads stats per month on npm registry
* [x] Project has public repository on famous online hosting platform (or been hosted on npm)

## Auto-update checklist

* [x] Has valid tags for each versions (for git auto-update)
* [ ] Auto-update setup
* [ ] Auto-update target/source is valid.
* [ ] Auto-update filemap is correct.

## Git commit checklist

* [ ] The first line of commit message is less then 50 chars; clean, clear and easy to understand.
* [x] The parent of the commit(s) in the PR is not older than 3 days.
* [x] Pull request is sent from a non-master branch with a meaningful name.
* [x] Separate unrelated changes into different commits.
* [x] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
* [x] Close corresponding issue in commit message
* [x] Mention related issue(s), people in commit message, comment.

### CODE REVIEW

**Comment by sashberd** [comment]
@PeterDaveHello @extend1994 @sufuf3 

1) I was not succeed to fetch all version with autoupdate npm tool, it was always fails with next message 

> error downloading 5.4.0 of firebase  it didn't exist: Error: read ECONNRESET
Most of latest versions were fetched by autoupdate config, thus I think it is fine

2) The `lib basepath` is for old versions 

3) versions 3.0.* was not fetched by auto-update config and have no idea why.. Need your help to review them

4) This lib has no one author, so the main contributor was mentioned in commit message. If there is a need to change or add another one give me to know

5) I added all *canary* versions to git ignore, I think it is good to exclude them

FYR

**Comment by sashberd** [comment]
@xtuc @MattIPv4 FYR this one. I hope auto update config is good enought

File: ajax/libs/firebase/package.json [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+      "files": [
+        "firebase*.+(js|map)"
+      ]
+    },
+    {
+      "basePath": "dist",
+      "files": [
+        "firebase*.+(js|map)"
+      ]
+    }
```
Comment (MattIPv4) [inline ajax/libs/firebase/package.json]: These seem to match nothing, are they for previous versions?

Comment (sashberd) [reply ajax/libs/firebase/package.json]: Yes, for version below 3.0.0

**Pull request merged**